### PR TITLE
Reduce cpu usage by making automatic file changes optional

### PR DESCRIPTION
**Pull request by negrifelipe**
Tested with some plugins installed:
- Feli.Economy
- Diffoz.Kits
- EvolutionPlugins.Dummy

Before: 
before 
After:
after

### CODE REVIEW

**Comment by negrifelipe** [comment]
> @negrifelipe please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@dotnet-policy-service agree

**Comment by negrifelipe** [comment]
@dotnet-policy-service agree

File: framework/OpenMod.Runtime/Runtime.cs [hunk 1]
```diff
@@ -219,8 +222,32 @@
...
+                                    break;
+                                default:
+                                    m_Logger!.LogWarning(
+                                        "Unknown config for 'reloadOnChange' in OpenMod configuration: {UnparsedConfig}",
+                                        unparsed);
+                                    break;
+
+                            }
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (negrifelipe) [reply framework/OpenMod.Runtime/Runtime.cs:247]: You really want a big one liner?

[the remark this answers is not in the document]
Comment (Trojaner) [reply framework/OpenMod.Runtime/Runtime.cs:247]: yes.

**Comment by MihailHavenT** [comment]
The changes are going well, but they give me errors in rocket plugins, and I don't know why...

[openmod-2023-10-15-01-27-59.log](https://github.com/openmod/openmod/files/12909121/openmod-2023-10-15-01-27-59.log)

**Comment by DiFFoZ** [comment]
I'm more preferring to use env value, just by setting `MONO_MANAGED_WATCHER` to `disabled`.
https://github.com/Unity-Technologies/mono/blob/unity-2021.3-mbe/mcs/class/System/System.IO/FileSystemWatcher.cs#L124

**Review by DiFFoZ — COMMENTED** [review]
See as above

**Comment by MihailHavenT** [comment]
When will the new patch with this update be released?

**Pull request closed without merging**